README.md

```diff
@@ -1,37 +1,50 @@
 
 ## Quick Usage
 
-In order to run MCTS, you must implement a `State` class which can fully describe the state of the world.  It must also implement four methods: 
+In order to run MCTS, you must implement a `State` class which can fully describe the state of the world.  It must also implement the following methods:
 
-- `getCurrentPlayer()`: Returns 1 if it is the maximizer player's turn to choose an action, or -1 for the minimiser player
+- `getCurrentPlayer()`: Returns 1 if it is the maximizer player's turn to choose an action, or -1 for the minimizer player
 - `getPossibleActions()`: Returns an iterable of all actions which can be taken from this state
 - `takeAction(action)`: Returns the state which results from taking action `action`
 - `isTerminal()`: Returns whether this state is a terminal state
-- `getReward()`: Returns the reward for this state.  Only needed for terminal states. 
+- `getReward()`: Returns the reward for this state: 0 for a draw, positive for a win by maximizer player or negative for a win by the minimizer player.  Only needed for terminal states.
 
-You must also choose a hashable representation for an action as used in `getPossibleActions` and `takeAction`.  Typically this would be a class with a custom `__hash__` method, but it could also simply be a tuple or a string.  
+You must also choose a hashable representation for an action as used in `getPossibleActions` and `takeAction`.  Typically this would be a class with a custom `__hash__` method, but it could also simply be a tuple or a string.
 
 Once these have been implemented, running MCTS is as simple as initializing your starting state, then running:
 
 ```python
 from mcts import mcts
 
-mcts = mcts(timeLimit=1000)
-bestAction = mcts.search(initialState=initialState)
+currentState = MyState()
+...
+searcher = mcts(timeLimit=1000)
+bestAction = searcher.search(initialState=currentState)
+currentState = currentState.takeAction(action)
+...
+
 ```
-See [naughtsandcrosses.py](https://github.com/pbsinclair42/MCTS/blob/master/naughtsandcrosses.py) for a simple example.  
+See [naughtsandcrosses.py](https://github.com/pbsinclair42/MCTS/blob/master/naughtsandcrosses.py) for a simple example.
+
+## Detailed usage
+A few customizations are possible through the `mcts` constructor:
+
+- The number of MCTS search rounds can be limited by either a given time limit or a given iteration number. 
+- The exploration constant $c$, which appears in the UCT score $w_i/n_i + c\sqrt{{ln N_i}/n_i}$ with theoretical default setting $c=\sqrt 2$, can be adapted to your game.
+- The default uniform random rollout/playout policy can be changed.
+
+A few statistics can be retrieved after each MCTS search call (see `naughtsandcrosses.py` example)
 
-## Slow Usage
-//TODO
+More of MCTS theory could be found at https://en.wikipedia.org/wiki/Monte_Carlo_tree_search and cited references.
```

Comment:
While this is useful, I'd still want to include more here before saying that the Detailed Usage / Slow Usage section is complete. Indeed, I feel like these notes would be useful included in the quick usage section too, so maybe pop them there instead and put the slow usage todo note back?

Reply:
Also, could we simplify this to the following:

When initialising the MCTS searcher, there are a few optional parameters that can be used to optimise the search:

- `timeLimit`: the maximum duration of the search in milliseconds. Exactly one of `timeLimit` and `iterationLimit` must be set.
- `iterationLimit`: the maximum number of search iterations to be carried out. Exactly one of `timeLimit` and `iterationLimit` must be set.
- `explorationConstant`: a weight used when searching to help the algorithm prioritise between exploring unknown areas vs deeper exploring areas it currently believes to be valuable. The higher this constant, the more the algorithm will prioritise exploring unknown areas. Default value is √2.
- `rolloutPolicy`: the policy to be used in the rollout phase when simulating one full playout. Default is a random uniform policy.

Reply:
Very good synthesis for the quick usage 👍 
Regarding the section "Slow usage", what do you mean by "slow" ? Do you intent "detailled information" ? Or have you in mind another mode for MCTS that indirectly slows down the search (like a different rolloutPolicy) ? Even if your material for this "slow usage" is not yet ready, could you either refine the title or put some hints in its TODO, if possible ?

Reply:
Aye, that was going to be for more detailed information on the whole system, so sure changing that header to "Detailed Information" would probably be clearer, go for it.